src/ringcentral-web-phone.js

```diff
@@ -1155,6 +1167,242 @@
 
     }
 
+    /*--------------------------------------------------------------------------------------------------------------------*/
+    //------------------QOS---------------------------------//
+
+
+    /**
+     * @this {SIP.Session}
+     */
+    function publish(options){
```

Comment:
Rename to `publishQosStats` or something similar.
Also throw an exception/fire event in case collection has not been started and there's nothing to publish